### PR TITLE
Update `etcd` certificates before updating any machine

### PR DESCRIPTION
**Pull request by ereslibre**
We need to include the new SAN on all the certificates before restarting
the first machine. Otherwise, this machine (a master) can find itself
isolated without being able to contact any etcd member with the name it
has (as the rest of the nodes haven't updated their certificates yet to
also include the new name on the SAN).

### CODE REVIEW

**Comment by ereslibre** [comment]
[Update to kubernetes 1.9](https://trello.com/c/E158VAoK/615-update-to-kubernetes-19)

File: salt/orch/update.sls [hunk 1]
```diff
@@ -83,6 +83,33 @@ pre-orchestration-migration:
...
+#
+# Let's force etcd to refresh certificates on all machines, restarting the etcd service so we can
+# continue with the upgrade, as certificates will be valid for the old and the new SAN.
+etcd-setup:
+  salt.state:
+    - tgt: 'roles:etcd'
+    - tgt_type: grain
+    - sls:
+      - etcd
+    - batch: 1
```
Comment (inercia) [inline salt/orch/update.sls]: Any reason for updating them one by one? Couldn't we run this step with `batch: {{ salt.caasp_etcd.get_cluster_size() }}`?

File: salt/orch/update.sls [hunk 2]
```diff
@@ -83,6 +83,33 @@ pre-orchestration-migration:
...
+# succeed on the orchestration if it doesn't have an etcd member. Assume M{1,2,3}, W{1,2}. Assume
+# etcd members are running on M2, W1 and W2.
+#
+# M1 updates its configurations on highstate and refers to the etcd nodes with the new names
+# (hostnames) instead of machine-ids, but M2, W1 and W2 still didn't run anything to refresh their
+# certificates and their etcd instances, what will make M1 fail because it cannot connect to any
+# etcd instance (as all certificates look invalid at this point for M2.hostname, W1.hostname and
+# W2.hostname). This makes the apiserver on M1 fail restarting itself until the orchestration reaches
+# M2 [there's no hard dependency on states between masters], but the orchestration already failed on
+# M1, so the global result will be failure nevertheless.
```
Comment (inercia) [inline salt/orch/update.sls]: Cound't we just update `/etc/hosts` in all the masters in the cluster prior to starting the update orchestration? The new `/etc/hosts` will contain both hostnames and machine-ids...

File: salt/orch/update.sls (hunk 1 above)
Comment (ereslibre) [reply salt/orch/update.sls]: It's better to do it safely so we don't restart all etcd instances at once. In any case, the `batch: 1` while being safer has negligible impact in terms of timing overall.

Comment (kiall) [reply salt/orch/update.sls]: That would mean all etcd's could restart at the same time - we want to keep the cluster up and functional during the upgrade.

File: salt/orch/update.sls (hunk 2 above)
Comment (ereslibre) [reply salt/orch/update.sls]: It wouldn't help, as the certificates on the running etcd members would still not contain the new name in the SANs, what would still isolate M1 when upgrading (in the presented case in the comment).

File: salt/orch/update.sls (hunk 1 above)
Comment (inercia) [reply salt/orch/update.sls]: Right.

File: salt/orch/update.sls (hunk 2 above)
Comment (inercia) [reply salt/orch/update.sls]: I see.

**Pull request merged**